packages/application-generic/src/health/queue-health-indicator.service.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  HealthCheckError,
+  HealthIndicator,
+  HealthIndicatorResult,
+} from '@nestjs/terminus';
+import { Injectable, Logger } from '@nestjs/common';
+
+import { QueueBaseService } from '../services';
+import { IHealthIndicator } from './health-indicator.interface';
+
+@Injectable()
+export abstract class QueueHealthIndicator
+  extends HealthIndicator
+  implements IHealthIndicator
+{
+  constructor(
+    private queueBaseService: QueueBaseService,
+    private indicatorKey: string,
+    private serviceName: string,
```

Comment:
Maybe let's think if some enums limiting the values that can be passed here would be something that would benefit us.

Reply:
Now sure I follow, could you elaborate on what you mean here or provide an example? :)